src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -154,6 +154,22 @@ void CoreWorkerMemoryStore::GetAsync(
   }
 }
 
+std::shared_ptr<RayObject> CoreWorkerMemoryStore::GetIfExists(const ObjectID &object_id) {
+  std::shared_ptr<RayObject> ptr;
+  {
+    absl::MutexLock lock(&mu_);
+    auto iter = objects_.find(object_id);
+    if (iter != objects_.end()) {
+      ptr = iter->second;
+    }
+  }
+  // It's important for performance to run the callback outside the lock.
```

Comment:
Hmm isn't this still prone to memory corruption? Why don't we make the bool for this function atomic?

Reply:
Yeah good point, let me make another PR to fix this existing issue.